### PR TITLE
Size some Map instances upfront

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: server/src/main/java/io/crate/execution/engine/collect/sources/SysTableRegistry.java [hunk 1]
```diff
@@ -1,61 +0,0 @@
...
-
-    public <R> void registerSysTable(TableInfo tableInfo,
-                                     Supplier<CompletableFuture<? extends Iterable<R>>> iterableSupplier,
-                                     Map<ColumnIdent, ? extends RowCollectExpressionFactory<R>> expressionFactories,
-                                     boolean involvesIO) {
-        RelationName ident = tableInfo.ident();
-        sysSchemaInfo.registerSysTable(tableInfo);
-        tableDefinitions.registerTableDefinition(
-            ident, new StaticTableDefinition<>(iterableSupplier, expressionFactories, involvesIO));
-    }
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/execution/engine/collect/sources/SysTableRegistry.java:59]: Probably deserves its own commit

Comment (mfussenegger) [reply server/src/main/java/io/crate/execution/engine/collect/sources/SysTableRegistry.java:59]: It's related to the sizing in that without this change it's not possible to pre-size the instances correctly.
